apps/client/app/components/PurchaseOrderForm.tsx

```diff
@@ -0,0 +1,167 @@
+// This is the only client component so far in the application due to the need for interactivity
+'use client'
+
+import React, { useEffect, useState } from 'react';
+import { useForm, Controller, useFieldArray } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import { PurchaseOrder, Item } from '../interfaces';
+
+// This component is passed in the Purchase Order and whether we are creating or editing
+interface PurchaseOrderFormProps {
+  purchaseOrder?: PurchaseOrder;
+  mode: 'create' | 'edit';
+}
+
+// Define the validation schema using Yup
+const validationSchema = Yup.object().shape({
+  id: Yup.number().required('Purchase Order is required'),
+  vendor_name: Yup.string().required('Vendor name is required'),
+  order_date: Yup.string().required('Order date is required'),
+  expected_delivery_date: Yup.string().required('Expected delivery date is required').test(
+    'is-greater',
+    'Expected delivery date must be after the order date',
+    function(value) {
+      const { order_date } = this.parent;
+      return new Date(order_date) < new Date(value);
+    }
+  ),
+  purchase_order_line_items: Yup.array().of(
+    Yup.object().shape({
+      id: Yup.number().required('Purchase Order Line Item is required'),
+      item_id: Yup.number().required('Item is required').positive(),
+      quantity: Yup.number().required('Quantity is required').positive().integer(),
+      unit_cost: Yup.number().required('Unit cost is required').positive(),
+    })
+  ).required(),
+  total_quantity: Yup.number().notRequired(),
+  total_cost: Yup.number().notRequired(),
+});
+
+// Fetch items to use in the dropdown of the purchase order line items
```

Comment:
This call get items to populate the purchase order detail items dropdown. I believe it would be better practice to move this to the CreatePurchaseOrder and EditPurchaseOrder server side rendered components and then pass the items as props to the PurchaseOrderFrom

Reply:
Implemented this change